.github/workflows/backward-compatibility-check.yml

```diff
@@ -0,0 +1,61 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+name: "Backward Compatibility Check"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "master"
+
+jobs:
+  backward-compatibility-check:
+    name: "Backward Compatibility Check"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        dependencies:
+          - "locked"
+        php-version:
+          - "7.4"
+        operating-system:
+          - "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+        with:
+          fetch-depth: 0
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "pcov"
+          php-version: "${{ matrix.php-version }}"
+          ini-values: memory_limit=-1
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v2"
+        with:
+          path: |
+            ~/.composer/cache
+            vendor
+          key: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
+          restore-keys: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
+
+      - name: "Install lowest dependencies"
```

Comment:
Note: while most of this is skipped, I kept the duplication across the file to easier diff them between each other.

This is hopefully (at some point) solved by github having a better DSL for github actions: YAML sucks badly for all this stuff, but there's no way around hipster culture, as it seems.